cram/cram_io.c

```diff
@@ -4575,3 +4575,27 @@ int cram_set_voption(cram_fd *fd, enum hts_fmt_option opt, va_list args) {
 
     return 0;
 }
+
+#define TEMPLATE_LEN 38
+int cram_check_EOF(cram_fd *fd)
+{
+    uint8_t buf[TEMPLATE_LEN];
+    const uint8_t template[TEMPLATE_LEN] = {
+        0x0f, 0x00, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff, 0x0f, 0xe0,
+        0x45, 0x4f, 0x46, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x05,
+        0xbd, 0xd9, 0x4f, 0x00, 0x01, 0x00, 0x06, 0x06, 0x01, 0x00,
+        0x01, 0x00, 0x01, 0x00, 0xee, 0x63, 0x01, 0x4b
+    };
+    if (CRAM_MAJOR_VERS(fd->version) < 2 ||
+        (CRAM_MAJOR_VERS(fd->version) == 2 && CRAM_MINOR_VERS(fd->version) == 0)) {
+        return 3; // No EOF support in cram versions less than 2.1
+    }
+    off_t offset = htell(fd->fp);
+    if (hseek(fd->fp, -TEMPLATE_LEN, SEEK_END) < 0) {
+        if (errno == ESPIPE) { hclearerr(fd->fp); return 2; }
+        else return -1;
+    }
+    if ( hread(fd->fp, buf, TEMPLATE_LEN) != TEMPLATE_LEN ) return -1;
+    if ( hseek(fd->fp, offset, SEEK_SET) < 0 ) return -1;
```

Comment:
This `( foo(…) )` parenthesis style is not used in nearby code.  So don't start :smile: